### PR TITLE
Update fastq_align_star for nf-test and handle transcriptome alignment better

### PR DESCRIPTION
**Pull request by pinin4fjords**
In this PR:

- Update fastq_align_star for nf-test
- Replace test data with RNA-seq data used by STAR module, that will actually align (test data used previously was not aligning). 
- Add analagous sorting, indexing etc for the transcriptome BAM that can be produced by STAR. 
- Supply a trancriptome fasta file that can be used by `samtools stats`

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by mashehu — CHANGES_REQUESTED** [review]
sorry, the last review wasn't supposed to be an approval, but a request for comments

File: subworkflows/nf-core/fastq_align_star/main.nf [hunk 1]
```diff
@@ -25,25 +30,49 @@ workflow FASTQ_ALIGN_STAR {
...
+    //
+
+    if (val_build_transcripts){
+        MAKE_TRANSCRIPTS_FASTA (
+            ch_fasta.map{it[1]},
+            ch_gtf.map{it[1]}
+        )
+        ch_transcripts_fasta = MAKE_TRANSCRIPTS_FASTA.out.transcript_fasta
+        ch_versions = ch_versions.mix(MAKE_TRANSCRIPTS_FASTA.out.versions)
+    }
```
[the remark this answers is not in the document]
Comment (pinin4fjords) [reply subworkflows/nf-core/fastq_align_star/main.nf:50]: I can just take this out and assume the user will supply the transcript fasta?

I'm basically just going for parity with what's done with the genomic alignments/ fasta, and added this for convenience.

[the remark this answers is not in the document]
Comment (mashehu) [reply subworkflows/nf-core/fastq_align_star/main.nf:50]: Sounds like a good compromise. if we see people having problem with that, we can add it.

File: subworkflows/nf-core/fastq_align_star/main.nf [hunk 2]
```diff
@@ -1,5 +1,8 @@
-include { STAR_ALIGN              } from '../../../modules/nf-core/star/align/main'
-include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'
+include { STAR_ALIGN                                                       } from '../../../modules/nf-core/star/align/main'
+include { RSEM_PREPAREREFERENCE as MAKE_TRANSCRIPTS_FASTA                  } from '../../../modules/nf-core/rsem/preparereference'
```
Comment (maxulysse) [inline subworkflows/nf-core/fastq_align_star/main.nf:2]: why using an alias?

Comment (pinin4fjords) [reply subworkflows/nf-core/fastq_align_star/main.nf:2]: Ported over from rnaseq. But don't worry- it's about to go :-)

File: subworkflows/nf-core/fastq_align_star/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinin4fjords) [reply subworkflows/nf-core/fastq_align_star/main.nf]: Done!

File: subworkflows/nf-core/fastq_align_star/meta.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ keywords:
   - genome
   - reference
 components:
+  - rsem/preparereference
```
Comment (mashehu) [inline subworkflows/nf-core/fastq_align_star/meta.yml:10]:
```suggestion
```

**Review by mashehu — APPROVED** [review]

**Comment by pinin4fjords** [comment]
thanks @mashehu !

**Pull request merged**